src/conn/conn_sweep.c

```diff
@@ -261,6 +261,67 @@ __sweep_server_run_chk(WT_SESSION_IMPL *session)
     return (FLD_ISSET(S2C(session)->server_flags, WT_CONN_SERVER_SWEEP));
 }
 
+/*
+ * __sweep_check_session_sweep --
+ *     Check for any "rogue" sessions, which did not run a session sweep in a long time.
+ */
+static void
+__sweep_check_session_sweep(WT_SESSION_IMPL *session, uint64_t now)
+{
+    WT_CONNECTION_IMPL *conn;
+    WT_SESSION_IMPL *s;
+    uint64_t last, last_cursor_big_sweep, last_sweep;
+    uint32_t i;
+
+    conn = S2C(session);
+
+    for (s = conn->sessions, i = 0; i < conn->session_cnt; ++s, ++i) {
+        /*
+         * Ignore inactive and internal sessions.
+         */
+        if (!s->active)
+            continue;
+        if (F_ISSET(s, WT_SESSION_INTERNAL))
+            continue;
+
+        last_cursor_big_sweep = s->last_cursor_big_sweep;
+        last_sweep = s->last_sweep;
+
+        last = last_cursor_big_sweep;
+        if (last_sweep > last)
+            last = last_sweep;
+        if (last == 0)
+            continue;
+
+        /*
+         * Check if the session did not run a sweep in 5 minutes. Handle the issue only once per
+         * violation.
+         */
+        if (last + 5 * 60 < now) {
+            if (!s->sweep_warning_5min) {
+                s->sweep_warning_5min = 1;
```

Comment:
I think we still want a statistic for this, even if we don't report it as a warning.  Sorry if I didn't communicate that.

Reply:
Oh that's right, sorry. I was deleting the warning and took down the stats for it as well; I'll put it back.